api/v1alpha3/conversion.go

```diff
@@ -139,6 +145,10 @@ func Convert_v1beta1_OpenStackClusterSpec_To_v1alpha3_OpenStackClusterSpec(in *i
 			Name: in.Bastion.Instance.IdentityRef.Name,
 		}
 	}
+
+	out.ManagedAPIServerLoadBalancer = in.APIServerLoadBalancer.Enabled
+	out.APIServerLoadBalancerAdditionalPorts = *(*[]int)(unsafe.Pointer(&in.APIServerLoadBalancer.AdditionalPorts))
```

Comment:
What's the reason for using unsafe here? couldn't it be as easy as the following?:

```suggestion
	out.APIServerLoadBalancerAdditionalPorts = in.APIServerLoadBalancer.AdditionalPorts
```

Reply:
I was too slow with answering, not its merged :(

I copied this from the previous autogenerated conversion.

The code is coming from `conversion-gen`: https://github.com/kubernetes/code-generator/blob/7e38d5771c5636f13555f0058e5936c521d2fe13/cmd/conversion-gen/generators/conversion.go#L929

Reply:
So this whole part is auto generated? If so we have to keep it, otherwise I would prefer to clean it up in a follow-up PR :-)

Reply:
Nope, my changes are not auto-generated, but I did not understand the semantics of the conversion code enough to remove the `unsafe.Pointer` logic provided by conversion-gen.